### PR TITLE
[PB-773]: Network download and decrypt modules

### PR DESCRIPTION
**Pull request by PixoDev**
This PR adds:

- Move endpoints for files and folders
- Improved code flow by unwrapping nil values correctly
- Download module and tests
- NetworkFacade download + decrypt tests

Note the decrypt module is just a proxy module for the AES decrypt with config from stream which is already testes

### CODE REVIEW

File: Sources/InternxtSwiftCore/Services/Network/Download.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+    
+    
+    func start(bucketId: String, fileId: String, destination: URL,  progressHandler: ProgressHandler? = nil,  debug: Bool = false) async throws ->  DownloadResult {
+        if bucketId.isValidHex == false {
+            throw DownloadError.InvalidBucketId
+        }
+        let info = try await networkAPI.getFileInfo(bucketId: bucketId, fileId: fileId)
+        
+        if info.shards.count > 1 {
```
Comment (sg-gs) [inline Sources/InternxtSwiftCore/Services/Network/Download.swift:62]: This is going to fail for uploads done before July - August 21'. Joining shards is not multipart (at least not typical multipart, as it does not use the common headers for that). If you receive multiple shards, just download and decrypt them (using the same decryptor) in order. In pseudocode: 

```
const decryptor = new DecryptStream();
const out = new FileSystemWriter();
for (const shard of shards) {
    const stream = getDownloadStream(shard)
   
    while (stream.isNotFinished) {
        decryptor.write(stream.read(<some_chunk_size>));
        out.write(decryptor.read(<some_chunk_size>));
     }
}
```

Comment (sg-gs) [reply Sources/InternxtSwiftCore/Services/Network/Download.swift:62]: Btw, I do not know if this scope should be tackled now, just saying.

File: Sources/InternxtSwiftCore/Services/Network/Encrypt.swift [hunk 2]
```diff
@@ -74,6 +73,8 @@ public struct Encrypt {
             sha256Hash.append(contentsOf: bytes)
         }
         
+        stream.close()
```
Comment (sg-gs) [inline Sources/InternxtSwiftCore/Services/Network/Encrypt.swift:76]: Should the hasher be closed, also?

File: Tests/InternxtSwiftCoreTests/Services/Network/DownloadTests.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                "created": "2023-08-16T10:51:34.592Z",
+                "renewal": "2023-11-14T10:51:34.592Z",
+                "mimetype": "application/octet-stream",
+                "filename": "9dac6039-7ffd-4c8b-ad39-4779e5b3b4b4",
+                "id": "64dcaa364a45bf00082ef12c",
+                "shards": [
+                    {
+                        "index": 0,
+                        "hash": "f754a1013f2fa5fe26ecbd9f159e81a616a89648",
+                        "url": "https://s3.gra.io.cloud.ovh.net/files/d6d26a2b-c013-491e-a4c3-6f09dfbd7cbd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=d0999b3fc01c4946a4b9f1dc49b22e9d%2F20230829%2Fgra%2Fs3%2Faws4_request&X-Amz-Date=20230829T172955Z&X-Amz-Expires=3600&X-Amz-Signature=6bad8c8b067b4d3704967c7b353dc3f29b1b67660ee91d74ef3e2451fad0a7fd&X-Amz-SignedHeaders=host"
```
Comment (sg-gs) [inline Tests/InternxtSwiftCoreTests/Services/Network/DownloadTests.swift:67]: Be careful here, although the link expires, we should avoid using real (temporal, object only, still) credentials here.

File: Sources/InternxtSwiftCore/Services/Network/Download.swift (hunk 1 above)
Comment (PixoDev) [reply Sources/InternxtSwiftCore/Services/Network/Download.swift:62]: I will tackle legacy downloads and multipart in a separated task so it doesn't block the main development

File: Sources/InternxtSwiftCore/Services/Network/Encrypt.swift (hunk 2 above)
Comment (PixoDev) [reply Sources/InternxtSwiftCore/Services/Network/Encrypt.swift:76]: The hasher is "done" once we call `.finalize()` if you attempt to update it after that, it throws an error, also note the hasher is not an stream

**Review by sg-gs — APPROVED** [review]

**Pull request merged**